cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java

```diff
@@ -140,14 +138,33 @@ public Integer doAction() throws Exception {
     boolean firstIteration = true;
     List<T> finalTos = new ArrayList<>();
     List<Object> generationInputs = new ArrayList<>();
-    for (File inputFile : this.inputFiles) {
+    for (Path inputFile : this.inputFiles) {
 
-      String extension = inputFile.getName().toLowerCase();
+      String extension = inputFile.getFileName().toString().toLowerCase();
       boolean isJavaInput = extension.endsWith(".java");
       boolean isOpenApiInput = extension.endsWith(".yaml") || extension.endsWith(".yml");
 
+      // checks for output root path and project root being detectable
+      if (this.outputRootPath == null && MavenUtil.getProjectRoot(inputFile, false) == null) {
+
+        LOG.info(
+            "Did not detect the input as part of a maven project, the root directory of the maven project was not found.");
+
+        // cancels with an error if more than one input file was provided
+        if (this.inputFiles.size() > 1) {
+          LOG.error(
+              "As multiple input files were provided, setting an automatic output path is not possible. Cancelling...");
+          System.exit(1);
+        }
+        LOG.info("Would you like to take '{}' as a root directory for output generation? \n"
+            + "type yes/y to continue or no/n to cancel (or hit return for yes).", System.getProperty("user.dir"));
```

Comment:
Why User dir?

Reply:
user.dir = working directory (path where JVM was run from)
user.home = home directory
But if pwd is more reliable, I can change it.